lib/libp2p_port.ex

```diff
@@ -688,4 +729,85 @@ defmodule LambdaEthereumConsensus.Libp2pPort do
       add_subscriber(state, topic, module)
     end)
   end
+
+  defp on_tick(time, %{genesis_time: genesis_time} = state) when time < genesis_time, do: state
+
+  defp on_tick(time, %{genesis_time: genesis_time, slot_data: slot_data} = state) do
+    # TODO: we probably want to remove this (ForkChoice.on_tick) from here, but we keep it
+    # here to have this serialized with respect to the other fork choice store modifications.
+
+    ForkChoice.on_tick(time)
+
+    new_slot_data = compute_slot(genesis_time, time)
+
+    updated_state = maybe_tick_validators(slot_data != new_slot_data, new_slot_data, state)
```

Comment:
nit: I may be a boomer here, but I feel that a simple if might be easier to follow, like so:

```elixir
if slot_data == new_slot_data do
  state
else
  updated_validators = notify_validators(state.validators, {:on_tick, new_slot_data})
  %{state | slot_data: new_slot_data, validators: updated_validators}
end
```

Reply:
Sure, I saw some `maybe_` prefixed functions and feel somewhat odd when variables are equated to an if expression xD `updated_state = if ...` but thats just me xD, already moved to the good ol' `if` ^^